hyperspace/BSONSerializer.cpp

```diff
@@ -175,6 +176,24 @@ void BSONSerializer::appendBooleanValue(const char *name, bool value)
     m_doc.append(value ? '\1' : '\0');
 }
 
+void BSONSerializer::appendArray(const char *name, QList<QVariant> value)
+{
+    m_doc.append(BSON_TYPE_ARRAY);
+
+    QVariantMap doc;
+    for (int i = 0; i < value.length(); i++) {
+        doc.insert(QString::number(i), value[i]);
+    }
+
+    BSONSerializer subDocument;
+    for (QVariantMap::const_iterator i = doc.constBegin(); i != doc.constEnd(); ++i) {
+        subDocument.appendValue(i.key().toLatin1().constData(), i.value());
+    }
```

Comment:
This doesn't make a lot of sense, you are copying all data into a new map and discarding it later, you should merge the two loops into just one loop.